### PR TITLE
PICARD 780: CAA cover art provider runs even if cover art has already been loaded

### PR DESCRIPTION
**Pull request by rahul-raturi**
I wonder why **old** was not returned in the first place. Without it, *next_in_queue* was unable to check for *WAIT* signal, as None was returned.

http://tickets.musicbrainz.org/browse/PICARD-780

### CODE REVIEW

File: picard/coverart/providers/__init__.py [hunk 1]
```diff
@@ -134,8 +134,8 @@ def queue_images(self):
         # CoverArtProvider.WAIT
         old = getattr(self, 'queue_downloads') #compat with old plugins
         if callable(old):
-            old()
             log.warning('CoverArtProvider: queue_downloads() was replaced by queue_images()')
+            return old()
```
Comment (zas) [inline picard/coverart/providers/__init__.py]: This is correct, though it should be in a different PR as it is unrelated to PICARD-780.

File: picard/coverart/providers/caa.py [hunk 2]
```diff
@@ -246,7 +246,8 @@ def _has_suitable_artwork(self):
 
     def enabled(self):
         """Check if CAA artwork has to be downloaded"""
-        if not super(CoverArtProviderCaa, self).enabled():
+        if not super(CoverArtProviderCaa, self).enabled() or \
+                self.coverart.front_image_found:
```
Comment (zas) [inline picard/coverart/providers/caa.py]: Did you test with releases having multiple front images in CAA ?

**Comment by rahul-raturi** [comment]
> Did you test with releases having multiple front images in CAA ?

No, I haven't. I tried several tracks but none of them had multiple front covers in the database. Though I'm still testing that, it would be helpful if someone else could confirm it.

> This is correct, though it should be in a different PR as it is unrelated to PICARD-780.

Erm, due to this picard was downloading a cover from amazon, even if fanart.tv plugin has sent WAIT and is downloading one (If you've enabled two providers fanart > amazon). Result was having multiple images in directory.

**Comment by zas** [comment]
> No, I haven't. I tried several tracks but none of them had multiple front covers in the database. Though I'm still testing that, it would be helpful if someone else could confirm it.

Be sure to test with such releases, this is where it is likely to break.

**Comment by zas** [comment]
> Erm, due to this picard was downloading a cover from amazon, even if fanart.tv plugin has sent WAIT and is downloading one (If you've enabled two providers fanart > amazon). Result was having multiple images in directory.

Ok, just a minor change anyway.

**Comment by rahul-raturi** [comment]
Patch is working fine with releases having multiple front images.

Coverarts at MB database: http://musicbrainz.org/release/bef90838-a09d-4853-ae85-6314d3ab5a4b/cover-art

Picard config:
picard config

Result:
picard output
